File: Day 2/Ques2.cpp
//question link:https://leetcode.com/problems/reverse-integer/
//solution:

class Solution {
public:
    long long int reverse(long long int x) {
         
        if(x<0){
            long long int digit;
            long long int rev=0;
            x=x*(-1);
            while(x>0){
                digit=x%10;
                rev=rev*10+digit;
                if(rev>2147483647){
                    return 0;
                }
                x=x/10;
            }
            return -rev;
        }
        else{
            long long int digit;
            long long int rev=0;
            while(x>0){
                digit=x%10;
                rev=rev*10+digit;
                if(rev>2147483647){
                    return 0;
                }
                x=x/10;
            }
            return rev;        
        }
    }
};

File: Day3/Ques2.py
#Question from Biweekly Contest. Link: https://leetcode.com/problems/count-nice-pairs-in-an-array/
class Solution:
       
    def countNicePairs(self, nums: List[int]) -> int:
        count=0
          
        for i in range(0,len(nums)):  
            
            for j in range(i+1,len(nums)):
                if i<j and j<len(nums):
                    rev1=0
                    x=nums[j]
                    while(x>0):
                        digit=x % 10
                        rev1=rev1*10+digit
                        x=int(x/10)
                    rev2=0
                    y=nums[i]
                    while(y>0):
                        digit=y % 10
                        rev2=rev2*10+digit
                        y=int(y/10)               
                    if(nums[i] + rev1) == (nums[j] + rev2):
                        count=count+1
        return count%1000000007
                        
             
